### PR TITLE
Pachyderm's PythonSDK

### PR DESCRIPTION
**Pull request by bbonenfant**
Creating a new PythonSDK that live in the core repository.

Design Doc: https://docs.google.com/document/d/1mZiStYydlioL4R9oqm5dhUYZm_nhgaIFywcwU0_9pIU

TODO before shipping
- [ ] Cleanup Readme
- [ ] Developer docs - how to get this running, run tests, etc
- [ ] Docs - docstrings, add to original grpc files and pass through, etc..
- [ ] Clean up proto generation?

other tasks...

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -406,6 +406,11 @@ validate-circle:
 	circleci config validate .circleci/main.yml
 	circleci config validate .circleci/config.yml
 
+python-sdk:
```
Comment (chainlink) [inline Makefile:409]: Let's create a makefile in `python-sdk/` and put this there

File: python-sdk/pachyderm_sdk/__init__.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    # >= 3.8
+    import importlib.metadata as metadata
+except ImportError:
+    #  < 3.8
+    import importlib_metadata as metadata  # type: ignore
+
+
+__pdoc__ = {"proto": False}
+
+__all__ = [
```
Comment (bbonenfant) [inline python-sdk/pachyderm_sdk/__init__.py:14]: This list needs to be cleaned up. Most of these aren't in the sdk

File: python-sdk/pachyderm_sdk/api/__init__.py [hunk 3]
```diff
@@ -0,0 +1 @@
+from .pfs import _additions as __pfs_additions
```
Comment (bbonenfant) [inline python-sdk/pachyderm_sdk/api/__init__.py:1]: This should be moved to the top level `__init__.py` Future work would be to revisit these additions

File: python-sdk/pachyderm_sdk/api/pfs/utils.py [hunk 4]
```diff
@@ -0,0 +1,20 @@
+from shutil import which
+from subprocess import run
+
+
+def check_pachctl(*, ensure_mount: bool = False) -> None:
```
Comment (bbonenfant) [inline python-sdk/pachyderm_sdk/api/pfs/utils.py:5]: This was used for the mount functionality which got removed. We technically do not need this anymore

File: python-sdk/pyproject.toml [hunk 5]
```diff
@@ -0,0 +1,37 @@
+[tool.poetry]
+name = "pachyderm_sdk"
+version = "0.0.1"
+description = "Python Pachyderm Client"
+authors = ["Pachyderm Integrations <integrations@pachyderm.io>"]
```
Comment (bbonenfant) [inline python-sdk/pyproject.toml:5]: should we change this now that we are hpe?

**Review by chainlink — APPROVED** [review]

**Pull request merged**